packages/connect-core/src/utils/transactions.ts

```diff
@@ -181,16 +116,13 @@ export async function buildForwardingFeePretransaction(
 
   if (feeDetails.tokenAddress && feeDetails.amount > BigInt(0)) {
     // Needs a token approval pretransaction
-    const forwardingTxWithTokenData: TransactionWithTokenData = {
-      ...forwardingTransaction,
-      token: {
-        address: feeDetails.tokenAddress,
-        spender: forwarderAddress, // since it's a forwarding transaction, always show the real spender
-        value: feeDetails.amount.toString(),
-      },
+    const tokenData: TokenData = {
```

Comment:
We no longer pass the Token details as part of the transaction object instead we provide a new object.